ale_linters/rst/rstcheck.vim

```diff
@@ -0,0 +1,29 @@
+" Author: John Nduli https://github.com/jnduli
+" Description: Rstcheck for reStructuredText files
+
+function! ale_linters#rst#rstcheck#Handle(buffer, lines) abort
+    " matches: 'bad_rst.rst:1: (SEVERE/4) Title overline & underline
+    " mismatch.'
+    let l:pattern = '\v^(.+):(\d*): \(([a-zA-Z]*)/\d*\) (.+)$'
+    let l:output = []
+    for l:match in ale#util#GetMatches(a:lines, l:pattern)
+        call add(l:output, {
+        \   'filename': l:match[1],
```

Comment:
Does `rstcheck` return relative paths or absolute paths?

If the paths are relative, you'll have to figure out what they are relative to, and then compute absolute paths from that with `ale#path#GetAbsPath`. If they are just relative to the working directory, prefix the command with `cd directory_for_file &&`. There are a few other linters that do this with the `ale#path#BufferCdString` function.